fog/view/enclave/impl/src/oblivious_utils.rs

```diff
@@ -0,0 +1,628 @@
+            ciphertext: vec![0u8; CLIENT_CIPHERTEXT_LENGTH],
+        })
+        .collect();
+
+    for shard_tx_out_search_result in shard_tx_out_search_results.iter() {
+        for client_tx_out_search_result in client_tx_out_search_results.iter_mut() {
+            maybe_overwrite_tx_out_search_result(
+                client_tx_out_search_result,
+                shard_tx_out_search_result,
+            );
+        }
+    }
+
+    Ok(client_tx_out_search_results)
+}
+
+fn maybe_overwrite_tx_out_search_result(
+    client_tx_out_search_result: &mut TxOutSearchResult,
+    shard_tx_out_search_result: &TxOutSearchResult,
+) {
+    let should_overwrite_tx_out_search_result = should_overwrite_tx_out_search_result(
+        client_tx_out_search_result,
+        shard_tx_out_search_result,
+    );
+    let shard_ciphertext_length = shard_tx_out_search_result.ciphertext.len();
+
+    let shard_cipher_text_length_delta = (CLIENT_CIPHERTEXT_LENGTH - shard_ciphertext_length) as u8;
+    // Need to add a 1 because the first byte is reserved for the delta.
+    assert!(
+        shard_cipher_text_length_delta >= 1,
+        "Shard ciphertext has unexpected length"
+    );
+    client_tx_out_search_result.ciphertext[0].conditional_assign(
+        &shard_cipher_text_length_delta,
+        should_overwrite_tx_out_search_result,
+    );
+    for idx in 0..shard_ciphertext_length {
+        // Offset the client ciphertext by 1 because the first byte is reserved for the
+        // length delta.
+        client_tx_out_search_result.ciphertext[idx + 1].conditional_assign(
+            &shard_tx_out_search_result.ciphertext[idx],
+            should_overwrite_tx_out_search_result,
+        );
+    }
+    client_tx_out_search_result.result_code.cmov(
+        should_overwrite_tx_out_search_result,
+        &shard_tx_out_search_result.result_code,
+    );
+}
+
+fn should_overwrite_tx_out_search_result(
+    client_tx_out_search_result: &TxOutSearchResult,
+    shard_tx_out_search_result: &TxOutSearchResult,
+) -> Choice {
+    let do_search_keys_match = client_tx_out_search_result
+        .search_key
+        .ct_eq(&shard_tx_out_search_result.search_key);
+
+    let client_tx_out_search_result_code = client_tx_out_search_result.result_code;
+    let shard_tx_out_search_result_code = shard_tx_out_search_result.result_code;
+
+    let client_code_is_found: Choice =
+        client_tx_out_search_result_code.ct_eq(&(TxOutSearchResultCode::Found as u32));
+    let client_code_is_not_found: Choice =
+        client_tx_out_search_result_code.ct_eq(&(TxOutSearchResultCode::NotFound as u32));
+
+    let shard_code_is_found: Choice =
+        shard_tx_out_search_result_code.ct_eq(&(TxOutSearchResultCode::Found as u32));
+
+    let shard_code_is_retryable_error =
+        is_code_retryable_error(shard_tx_out_search_result.result_code);
+    let shard_code_is_bad_search_key =
+        shard_tx_out_search_result_code.ct_eq(&(TxOutSearchResultCode::BadSearchKey as u32));
+
```

Comment:
it would be nice to have a comment just like trying to explain at a high level what we want to happen:

* we make the same query to several shards and get several responses
* we want to prioritize "found" responses, and then beyond that, the "bad search key", which means the argument was invalid.
* after that the other two errors are like "retriable" errors that the client will probably just wait a while and retry
* "not found" response gets overwritten by  any other response

Reply:
Done.